src/rgw/rgw_sync_module_pubsub.cc

```diff
@@ -1265,25 +1269,35 @@ class RGWPSHandleObjEventCR : public RGWCoroutine {
               record->configurationId = sub->sub_conf->s3_id;
               yield call(PSSubscription::store_event_cr(sync_env, sub, record));
               if (retcode < 0) {
+                if (perfcounter) perfcounter->inc(l_rgw_pubsub_store_fail);
                 ldout(sync_env->cct, 1) << "ERROR: failed to store record for subscription=" << *siter << " ret=" << retcode << dendl;
               } else {
+                if (perfcounter) perfcounter->inc(l_rgw_pubsub_store_ok);
                 event_handled = true;
               }
               if (sub->sub_conf->push_endpoint) {
                   ldout(sync_env->cct, 20) << "push record for subscription=" << *siter << " owner=" << *oiter << " ret=" << retcode << dendl;
                 yield call(PSSubscription::push_event_cr(sync_env, sub, record));
                 if (retcode < 0) {
+                  if (perfcounter) perfcounter->inc(l_rgw_pubsub_push_failed);
                   ldout(sync_env->cct, 1) << "ERROR: failed to push record for subscription=" << *siter << " ret=" << retcode << dendl;
                 } else {
+                  if (perfcounter) perfcounter->inc(l_rgw_pubsub_push_ok);
                   event_handled = true;
                 }
               }
             }
           }
           if (!sub_conf_found) {
             // could not find conf for subscription at user or global levels
+            if (perfcounter) perfcounter->inc(l_rgw_pubsub_missing_conf);
             ldout(sync_env->cct, 1) << "ERROR: failed to find subscription config for subscription=" << *siter 
               << " ret=" << last_sub_conf_error << dendl;
+              if (retcode == -ENOENT) {
+                // missing subscription info should be reflected back as invalid argument
+                // and not as missing object
+                retcode =  -EINVAL;
+              }
```

Comment:
so this is the error we return back through the sync module to data sync? data sync will continuously retry on any error code (except the special case for ENOENT) - is that what we really want here? is it likely that a subscription will show up after the fact to resolve this? or should we just log the error and return success to move on?

Reply:
this is not a case of missing configuration of thew subscription. if not configuration exist, this code will not be called. this may happen when:
* race condition - subscription configuration deleted while sync in process. next retry of the object will not find this subscription anyway
* temporary I/O issue - hopefully resolved the next time

We may optimize the flow for the 2 cases, if we can easily distinguish between them.
But, IMO, not needed, especially that there are other failures (I didn't think about) that may benefit from a retry.

Note, that before the fix in line 1238, there was always a failure when trying to fetch either the user or global level confs (where retries were pointless).